src/DynamoCore/Models/PresetsModel.cs

```diff
@@ -0,0 +1,190 @@
+            designStates.Add(loadedState);
+        }
+        #endregion
+
+        # region properties
+        public IEnumerable<PresetState> DesignStates { get { return designStates; } }
+        #endregion
+
+        #region constructor
+        public PresetsModel()
+        {
+            designStates = new List<PresetState>();
+        }
+        #endregion
+
+        #region serialization / deserialzation
+        // we serialze the presets to xml like a model, but we deserialze them before the workspacemodel is constructed
+        //during save and load of the graph, can just inject this into workspacemodel 
+        //when a new designstate is created we'll serialize all the current nodes into a new xmlelement
+        //but not actually write this xml to a file until the graph is saved
+
+        //grabbed some methods needed from modelbase for serialization
+        protected virtual XmlElement CreateElement(XmlDocument xmlDocument, SaveContext context)
+        {
+            string typeName = GetType().ToString();
+            XmlElement element = xmlDocument.CreateElement(typeName);
+            return element;
+        }
+
+        public XmlElement Serialize(XmlDocument xmlDocument, SaveContext context)
+        {
+            var element = CreateElement(xmlDocument, context);
+            SerializeCore(element, context);
+            return element;
+        }
+
+        protected virtual void SerializeCore(System.Xml.XmlElement element, SaveContext context)
+        {
+
+            foreach (var state in designStates)
+            {
+                var parent = element.OwnerDocument.CreateElement("PresetState");
+                element.AppendChild(parent);
+                parent.SetAttribute("Name", state.Name);
+                parent.SetAttribute("Description", state.Description);
+                parent.SetAttribute("guid", state.Guid.ToString());
+                //the states are already serialized
+                foreach (var serializedNode in state.SerializedNodes)
+                {
+                    //need to import the node to cross xml contexts
+                    var importNode = parent.OwnerDocument.ImportNode(serializedNode, true);
+                    parent.AppendChild(importNode);
+                }
+
+            }
+        }
+
+        internal static PresetsModel LoadFromXml(XmlDocument xmlDoc, NodeGraph nodegraph,ILogger logger)
+        {
+            var loadedStateSet = new PresetsModel();
+
+            //create a new state inside the set foreach state present in the xmldoc
+
+            foreach (XmlElement element in xmlDoc.DocumentElement.ChildNodes)
+            {
+                if (element.Name == typeof(PresetsModel).ToString())
+                {
+                    foreach (XmlElement stateNode in element.ChildNodes)
+                    {
+                        var name = stateNode.GetAttribute("Name");
+                        var des = stateNode.GetAttribute("Description");
```

Comment:
It's a good idea to make attribute names like these `public const`, rather than putting them inline in the code.